### PR TITLE
Add RateLimitError to jsonrpc

### PR DESCRIPTION
**Pull request by mkondratek**
Issue: https://github.com/sourcegraph/sourcegraph/issues/56455.

Jetbrains part: https://github.com/sourcegraph/jetbrains/pull/111

## Test plan
mock throw the error from the autocompletion and recipe request
runide with a local clone of cody

### CODE REVIEW

File: vscode/src/jsonrpc/agent-protocol.ts [hunk 1]
```diff
@@ -123,6 +124,8 @@ export interface SelectedCompletionInfo {
     readonly text: string
 }
 export interface AutocompleteResult {
+    // todo: export enum ExecuteResultStatus
+    status: ExecuteResultStatus
```
Comment (mkondratek) [inline vscode/src/jsonrpc/agent-protocol.ts:128]: Fun fact:
In `cody-agent-trace.json`, the enum field is represented as a number:
```
[Trace - 00:40:55 AM] Received response 'recipes/execute - (12)' in 3814ms
Result: {
  "status": 0
}
Error: null
```
However, in the jetbrains repo it is read correctly as the corresponding enum type.

**Comment by jdorfman** [comment]
Thanks @mkondratek I will make sure someone checks this out!

**Comment by abeatrix** [comment]
@jdorfman  Im not familiar with the agent part of the codebase, so it's probably best to tag someone else from the jetBrains team to verify. @sourcegraph/jetbrains-plugin ?

**Comment by jdorfman** [comment]
@abeatrix my bad I'm going to move this to JetBrains repo

**Comment by Strum355** [comment]
Fyi this is one of the VirtusLabs contractors from the shared slack channel, not a random contributor

**Review by Strum355 — APPROVED** [review]

**Pull request merged**